grpc/wrappers.go

```diff
@@ -583,3 +558,865 @@ func (ras *RegistrationAuthorityServerWrapper) AdministrativelyRevokeCertificate
+	}
+
+	if response == nil || response.Serial == nil || response.SubscriberApproved == nil || response.Status == nil || response.OcspLastUpdated == nil || response.RevokedDate == nil || response.RevokedReason == nil || response.LastExpirationNagSent == nil || response.OcspResponse == nil || response.NotAfter == nil || response.IsExpired == nil {
+		fmt.Println("DO DOO")
+		fmt.Printf("%#v\n", response)
+		return core.CertificateStatus{}, errIncompleteResponse
+	}
+
+	return core.CertificateStatus{
+		Serial:                *response.Serial,
+		SubscriberApproved:    *response.SubscriberApproved,
+		Status:                core.OCSPStatus(*response.Status),
+		OCSPLastUpdated:       time.Unix(0, *response.OcspLastUpdated),
+		RevokedDate:           time.Unix(0, *response.RevokedDate),
+		RevokedReason:         revocation.Reason(*response.RevokedReason),
+		LastExpirationNagSent: time.Unix(0, *response.LastExpirationNagSent),
+		OCSPResponse:          response.OcspResponse,
+		NotAfter:              time.Unix(0, *response.NotAfter),
+		IsExpired:             *response.IsExpired,
+	}, nil
+}
+
+func (sac StorageAuthorityClientWrapper) CountCertificatesRange(ctx context.Context, earliest, latest time.Time) (int64, error) {
+	localCtx, cancel := context.WithTimeout(ctx, sac.timeout)
+	defer cancel()
+
+	earliestNano := earliest.UnixNano()
+	latestNano := latest.UnixNano()
+
+	response, err := sac.inner.CountCertificatesRange(localCtx, &sapb.Range{
+		Earliest: &earliestNano,
+		Latest:   &latestNano,
+	})
+	if err != nil {
+		return 0, unwrapError(err)
+	}
+
+	if response == nil || response.Count == nil {
+		return 0, errIncompleteResponse
+	}
+
+	return *response.Count, nil
+}
+
+func (sac StorageAuthorityClientWrapper) CountCertificatesByNames(ctx context.Context, domains []string, earliest, latest time.Time) (map[string]int, error) {
+	localCtx, cancel := context.WithTimeout(ctx, sac.timeout)
+	defer cancel()
+
+	earliestNano := earliest.UnixNano()
+	latestNano := latest.UnixNano()
+
+	response, err := sac.inner.CountCertificatesByNames(localCtx, &sapb.CountCertificatesByNamesRequest{
+		Names: domains,
+		Range: &sapb.Range{
+			Earliest: &earliestNano,
+			Latest:   &latestNano,
+		},
+	})
+	if err != nil {
+		return nil, unwrapError(err)
+	}
+
+	if response == nil || response.CountByNames == nil {
+		return nil, errIncompleteResponse
+	}
+
+	names := make(map[string]int, len(response.CountByNames))
+	for _, element := range response.CountByNames {
+		if element == nil || element.Name == nil || element.Count == nil {
+			return nil, errIncompleteResponse
+		}
+		names[*element.Name] = int(*element.Count)
+	}
+
+	return names, nil
+}
+
+func (sac StorageAuthorityClientWrapper) CountRegistrationsByIP(ctx context.Context, ip net.IP, earliest, latest time.Time) (int, error) {
+	localCtx, cancel := context.WithTimeout(ctx, sac.timeout)
+	defer cancel()
+
+	earliestNano := earliest.UnixNano()
+	latestNano := latest.UnixNano()
+	ipStr := ip.String()
```

Comment:
Here we serialize `net.IP` as a string; elsewhere (DB) we do it as bytes. I *think* it makes more sense to be consistent about using bytes?